qkrehdwns/likelion-assignment04/src/main/java/likelion/likelionassignment04/service/CityService.java

```diff
@@ -0,0 +1,65 @@
+package likelion.likelionassignment04.service;
+
+
+import likelion.likelionassignment04.domain.City;
+import likelion.likelionassignment04.domain.Country;
+import likelion.likelionassignment04.dto.city.CityResponseDto;
+import likelion.likelionassignment04.dto.city.CitySaveDto;
+import likelion.likelionassignment04.repository.CityRepository;
+import likelion.likelionassignment04.repository.CountryRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CityService {
+
+    private final CityRepository cityRepository;
+    private final CountryRepository countryRepository;
+
+    // 도시 저장
+    @Transactional
+    public void saveCity(CitySaveDto citySaveDto) {
+       Country country = countryRepository.findById(citySaveDto.country_id())
+           .orElseThrow(IllegalArgumentException::new);
+
+        City city = City.builder()
+            .city_name(citySaveDto.city_name())
+            .country(country)
+            .build();
```

Comment:
지금처럼 서비스 계층에서 직접 엔티티를 생성하는 방법도 괜찮지만
객체 생성 책임은 서비스보다는 도메인에게 부여했을 때 좀 더 객체지향스러운 코드가 될 수 있어요-!

```
public static City create(String cityName, Country country) {
    return City.builder()
            .city_name(cityName)
            .country(country)
            .build();
}
```

이런 식으로요!

```suggestion
        City city = City.create(citySaveDto.city_name(), country);
```

아래 첨부한 블로그에 나와있는 것처럼
"**도메인 객체에 정보만 담겨 있고 아무런 기능도 없는 것은 온전한 객체라고 볼 수 없다**"는 점을 기억하시면서 코드를 작성해보시면 좀 더 객체지향적인 코드로 발전할 수 있을 것 같아요!

[도메인 주도 설계 관련 블로그](https://mangkyu.tistory.com/160)

Reply:
피드백 감사합니다! 
글을 읽어보고 앞으로 좀 더 객체 지향적인 코드를 만들어 보도록 하겠습니다!